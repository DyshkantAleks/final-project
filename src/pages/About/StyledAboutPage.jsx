import styled from "styled-components";
import {device} from "../../styles/breakpoints/breakpoints";

export const ContactLink = styled.a`
font-size: 1.4rem;
line-height: 1.5;
text-decoration: underline;
color: inherit;
`;

export const TextContainer = styled.div`
flex-basis: 60%;
margin-right: 5rem;
`;
export const ContainerPage = styled.div`
display: flex;
flex-direction: column;
@media${device.tabletM}{
justify-content: space-between;
flex-direction: row;
}
`;
export const ContainerContactForm = styled.div`
border: .1rem solid #aecfd9;
box-shadow: 5px 5px 10px #aecfd9 ;
padding: 2rem;
@media${device.tabletM}{
width: 30rem;
margin: 0 auto;
align-self: flex-start;
}
@media${device.tabletM}{
margin: 0;
}
`;
export const ContactForm = styled.p`
font-size: 1.4rem;
line-height: 1.5;
font-weight: 500;
@media${device.tabletM}{

}
`;


export const Text = styled.p`
font-size: 1.4rem;
line-height: 1.5;
`

export const TextList = styled.ul`
font-size: 1.4rem;
line-height: 1.5;


`
export const TextListItem = styled.li`

`