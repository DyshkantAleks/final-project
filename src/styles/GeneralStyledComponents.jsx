import styled from 'styled-components';
import { device } from './breakpoints/breakpoints';

export const ContentContainer = styled.div`
    padding: 0 1.6rem;
    margin: 0 auto;
    
    @media ${device.tabletM}{
        padding: 0 2.4rem;
    }
    @media ${device.tabletL}{
        padding: 0 5rem;
    }
    
    @media ${device.desktop}{
        padding: 4rem 0;
        width: 120rem;
    }
`