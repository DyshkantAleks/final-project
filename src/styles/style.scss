:root {
  font-size: 10px !important; //надо important заменить при сборке
}

body {
  box-sizing: border-box;
  width: 100%;
  min-width: 320px;
  margin: 0;
  padding: 0;
  font-family: 'Montserrat', sans-serif !important;

  .ant-tooltip {
    z-index: 1200;
  }
}

a:hover {
  color: initial !important;
}