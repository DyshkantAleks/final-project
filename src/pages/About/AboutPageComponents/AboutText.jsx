import React from 'react';

import { ContentContainer } from '../../../styles/GeneralStyledComponents';
import { ContainerPage, TextContainer, Text, TextList, TextListItem } from '../StyledAboutPage';
import { ContactsForm } from './ContactForm';

export const AboutText = () => {
  return (
    <ContentContainer>
      <ContainerPage>
        <TextContainer>
          <h2>МЫ - КОМАНДА МОЛОДЫХ УВЛЕЧЕННЫХ ПРОФЕССИОНАЛОВ</h2>
          <Text>Мы уверены, что пространство, в котором мы живем, должно отражать наши мечты и стремления, и каждый из
            нас
            в силах создать достойный восхищения интерьер, в котором приятно находиться и в который всегда хочется
            возвращаться.
          </Text>
          <Text>Наш опыт, стремление к комплексному подходу во всем, уважение к желаниям клиента позволяют нам
            помогать
            Вам реализовать любые ваши идеи в реальные сроки за разумные деньги.

          </Text>
          <Text>Мы работаем напрямую с лучшими фабриками Европы и Азии, производящими мебель и предметы интерьера
            ценового
            сегмента «средний+», «высокий», «премиум», что позволяет нам предлагать Вам актуальные новинки по
            конкурентным
            ценам. Большой выбор товаров в наличии и под заказ позволяют нам помочь Вам подобрать достойные варианты в
            рамках Вашего бюджета.
          </Text>
          <Text>Мы уже реализовали ряд проектов, основываясь на принципах:</Text>
          <TextList>
            <TextListItem>Комплексный подход эффективен всегда.</TextListItem>
            <TextListItem>Нереализуемых задач не бывает. Мы всегда готовы предложить варианты решения даже самой
              нетрадиционной задачи.
            </TextListItem>
            <TextListItem>Важен баланс качества результата, сроков реализации проекта и бюджета.</TextListItem>
          </TextList>
        </TextContainer>
        <ContactsForm />
      </ContainerPage>
    </ContentContainer>
  )
};