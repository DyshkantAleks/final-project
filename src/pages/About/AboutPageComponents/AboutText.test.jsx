import React from 'react';
import { shallow } from 'enzyme';

import { ContentContainer } from '../../../styles/GeneralStyledComponents';
import { ContainerPage, Text, TextContainer, TextList, TextListItem } from '../StyledAboutPage';
import { AboutText } from './AboutText';

const setUp = (props = {}) => shallow(<AboutText {...props} />);

describe('AboutText component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should be render with general wrapper', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.length).toEqual(1);
  });
  it('should be render with wrapper', () => {
    const wrapper = component.find(ContainerPage);
    expect(wrapper.length).toEqual(1);
  });
  it('should be render with text container', () => {
    const textContainer = component.find(TextContainer);
    expect(textContainer.length).toEqual(1);
  });
  it('should be render with <h2>', () => {
    const title = component.find('h2');
    expect(title.length).toEqual(1);
    expect(title.text()).toEqual('МЫ - КОМАНДА МОЛОДЫХ УВЛЕЧЕННЫХ ПРОФЕССИОНАЛОВ');
  });
  it('should be render with text', () => {
    const text = component.find(TextContainer).find(Text);
    expect(text.length).toEqual(4);
    expect(text.at(0).text()).toEqual('Мы уверены, что пространство, в котором мы живем, должно отражать наши мечты и стремления, и каждый из нас в силах создать достойный восхищения интерьер, в котором приятно находиться и в который всегда хочется возвращаться.');
    expect(text.at(1).text()).toEqual('Наш опыт, стремление к комплексному подходу во всем, уважение к желаниям клиента позволяют нам помогать Вам реализовать любые ваши идеи в реальные сроки за разумные деньги.');
    expect(text.at(2).text()).toEqual('Мы работаем напрямую с лучшими фабриками Европы и Азии, производящими мебель и предметы интерьера ценового сегмента «средний+», «высокий», «премиум», что позволяет нам предлагать Вам актуальные новинки по конкурентным ценам. Большой выбор товаров в наличии и под заказ позволяют нам помочь Вам подобрать достойные варианты в рамках Вашего бюджета.');
    expect(text.at(3).text()).toEqual('Мы уже реализовали ряд проектов, основываясь на принципах:');
  });
  it('should be render with <ul>', () => {
    const textList = component.find(TextList);
    expect(textList.length).toEqual(1);
  });
  it('should render with list', () => {
    const textListItem = component.find(TextListItem);
    expect(textListItem.length).toEqual(3);
    expect(textListItem.at(0).text()).toEqual('Комплексный подход эффективен всегда.');
    expect(textListItem.at(1).text()).toEqual('Нереализуемых задач не бывает. Мы всегда готовы предложить варианты решения даже самой нетрадиционной задачи.');
    expect(textListItem.at(2).text()).toEqual('Важен баланс качества результата, сроков реализации проекта и бюджета.');
  });
  it('snapshot <AboutText />', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.debug()).toMatchSnapshot();
  });
});