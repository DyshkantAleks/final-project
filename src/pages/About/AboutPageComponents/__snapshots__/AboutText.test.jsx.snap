// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutText component snapshot <AboutText /> 1`] = `
"<styled.div>
  <styled.div>
    <styled.div>
      <h2>
        МЫ - КОМАНДА МОЛОДЫХ УВЛЕЧЕННЫХ ПРОФЕССИОНАЛОВ
      </h2>
      <styled.p>
        Мы уверены, что пространство, в котором мы живем, должно отражать наши мечты и стремления, и каждый из нас в силах создать достойный восхищения интерьер, в котором приятно находиться и в который всегда хочется возвращаться.
      </styled.p>
      <styled.p>
        Наш опыт, стремление к комплексному подходу во всем, уважение к желаниям клиента позволяют нам помогать Вам реализовать любые ваши идеи в реальные сроки за разумные деньги.
      </styled.p>
      <styled.p>
        Мы работаем напрямую с лучшими фабриками Европы и Азии, производящими мебель и предметы интерьера ценового сегмента «средний+», «высокий», «премиум», что позволяет нам предлагать Вам актуальные новинки по конкурентным ценам. Большой выбор товаров в наличии и под заказ позволяют нам помочь Вам подобрать достойные варианты в рамках Вашего бюджета.
      </styled.p>
      <styled.p>
        Мы уже реализовали ряд проектов, основываясь на принципах:
      </styled.p>
      <styled.ul>
        <styled.li>
          Комплексный подход эффективен всегда.
        </styled.li>
        <styled.li>
          Нереализуемых задач не бывает. Мы всегда готовы предложить варианты решения даже самой нетрадиционной задачи.
        </styled.li>
        <styled.li>
          Важен баланс качества результата, сроков реализации проекта и бюджета.
        </styled.li>
      </styled.ul>
    </styled.div>
    <ContactsForm />
  </styled.div>
</styled.div>"
`;
